src/dynamic_array.c

```diff
@@ -47,3 +45,37 @@ void iter_seq(DynamicArray *dy_arr) {
         printf("%d", dy_arr->data[i]);
     };
 };
+
+void insert_at(DynamicArray *dy_arr, size_t pos, int value) {
+    if (dy_arr->size == dy_arr->capacity) {
+        resize(dy_arr);
+    };
+
+    if (pos > dy_arr->size) {
+        fprintf(stderr,
+                "Specified index is greater than current array size.\n");
+        exit(EXIT_FAILURE);
+    };
+
+    memmove(dy_arr->data + pos + 1, dy_arr->data + pos,
+            sizeof(int) * (dy_arr->size - pos));
+    dy_arr->size = dy_arr->size + 1;
+    set_at(dy_arr, pos, value);
+};
+
+// What I'd love to see are 3 specific cases:
+//(1) Use an initial size of 0.
+//(2) Insert something in front of the first element.
+//(3) Insert something after the last element.
+
+void resize(DynamicArray *dy_arr) {
+    size_t new_capacity = GROWTH_FACTOR * dy_arr->capacity + 1;
```

Comment:
use empy array check. If dy_arr->capacity==0, set it as 2 (like default std::vector init capacity value in C++).
otherwise, if dy_arr->capacity!=0, new_capacity = GROWTH_FACTOR * dy_arr->capacity. (in test case test_insert_at_1 line: assert(dy_arr.capacity == 1); easy to be equal 2 ^_^